CMakeLists.txt

```diff
@@ -20,18 +20,30 @@ cmake_minimum_required(VERSION 3.6)
 project(hypervisor)
 
 # ------------------------------------------------------------------------------
-# Cmake global properties
+# Config
 # ------------------------------------------------------------------------------
 
 include(scripts/cmake/config/default.cmake)
+include_external_config()
```

Comment:
I think this needs to be called before the default configs are included, or otherwise moved to an appropriate location high up in the default.cmake config file. Reason is that some of the "default values" rely on the values of other configs. In order for user-defined values to play nicely with the default values (and the add_config macro), they need to be defined first.